src/assistants/openai/openai_assistant.rs

```diff
@@ -136,14 +138,70 @@ impl OpenAIAssistant {
         Ok(())
     }
 
-    /*
-     * This function performs all the orchestration needed to submit a prompt and get and answer
-     */
+    ///
+    /// This function performs all the orchestration needed to submit a prompt and get and answer
+    ///
     pub async fn get_answer<T: JsonSchema + DeserializeOwned>(
         &mut self,
         message: &str,
         file_ids: &[String],
     ) -> Result<T> {
+        // Instruct the Assistant to answer with the right Json format
+        // Output schema is extracted from the type parameter
+        let schema = schema_for!(T);
+
+        // Convert the schema to a JSON value
+        let mut schema_json: Value = serde_json::to_value(&schema)?;
+
+        // Remove '$schema' and 'title' elements that are added by schema_for macro but are not needed
```

Comment:
`schema_for!` macro includes fields `$schema` and `title` that sometimes confuse AI.